dev_tools/Dockerfile

```diff
@@ -11,14 +21,24 @@ ENV LANGUAGE en_US:en
 ENV LC_ALL en_US.UTF-8 
 
 # Make python3 default
-RUN rm -f /usr/bin/python && ln -s /usr/bin/python3 /usr/bin/python
+RUN rm -f /usr/bin/python \
+     && ln -s /usr/bin/python3 /usr/bin/python
 
 # Get a current copy of Cirq from Github.
 RUN git clone https://github.com/quantumlib/Cirq
 
+COPY requirements.txt .
+COPY Cirq/cirq/contrib/contrib-requirements.txt .
+COPY Cirq/dev_tools/conf/pip-list-dev-tools.txt .
+
+RUN cat requirements.txt contrib-requirements.txt pip-list-dev-tools.txt >> dependencies.txt \
+     && pip3 install --user -r dependencies
+
 # Install the needed Python libraries.
-RUN pip3 install -r Cirq/requirements.txt -r Cirq/cirq/contrib/contrib-requirements.txt -r Cirq/dev_tools/conf/pip-list-dev-tools.txt
+RUN pip3 install --user .
 
-WORKDIR /Cirq
+FROM python:3.7-slim AS build-image
+COPY --from=compile-image /root/.local /root/.local
```

Comment:
thinking about this - it might be problematic to do the multistage build this way because the final image will be missing all the visualization libraries that we apt-get installed :/ Can you try to run an example or a notebook that uses matplotlib? You should be able to run `docker run -p 8888:8888 cirq-image jupyter notebook --port=8888 --no-browser --ip=0.0.0.0` and then connect to it via http://localhost:8888

Reply:
I see what you're saying. 
I'm getting this error:
<details>
<summary>Error</summary>
<p>
docker: Error response from daemon: OCI runtime create failed: container_linux.go:349: starting container process caused "exec: \"jupyter\": executable file not found in $PATH": unknown.
</p>
</details>

When running the file. What is the correct way to handle this?

Reply:
Figured it out. Jupyter is not installed by default on python base image.

Reply:
@balopat I made some changes and I was actually just able to run a simple example with matplotlib successfully in my browser. I then ran the notebook at `docs/tutorials/basics.ipynb` and everything worked as well.

Reply:
Hey @seunomonije - nice! Thank you for pushing through with this, yes jupyter was missing. 
No need to open a fresh PR, you can just convert this one into a non-draft PR - it's better to have all the conversation on the same PR, so when questions arise we can come back to these decisions.